src/d3d8/d3d8_device.cpp

```diff
@@ -119,28 +120,22 @@ namespace dxvk {
         return E_FAIL;
     }
 
-    if (unlikely(FAILED(res)))
-      goto done;
-    
-    // Immediately issue the query.
-    // D3D9 will begin it automatically before ending.
-    res = pQuery->Issue(D3DISSUE_END);
-    if (unlikely(FAILED(res))) {
-      goto done;
-    }
-
-    // TODO: Will immediately issuing the query without doing any API calls
-    // actually yield meaingful results? And should we flush or let it mellow?
-    res = pQuery->GetData(pDevInfoStruct, DevInfoStructSize, D3DGETDATA_FLUSH);
-
-  done:
-    if (unlikely(FAILED(res))) {
+    if (FAILED(res)) {
       if (res == D3DERR_NOTAVAILABLE) // unsupported
         return E_FAIL;
       else // any unknown error
         return S_FALSE;
     }
 
+    if (pQuery != nullptr) {
+      // Immediately issue the query. D3D9 will begin it automatically before ending.
+      pQuery->Issue(D3DISSUE_END);
+      // TODO: Will immediately issuing the query actually yield meaingful results?
+      // Only relevant once RESOURCEMANAGER or VERTEXSTATS are implemented by D9VK,
+      // since VCACHE queries will immediately return data during this call.
+      res = pQuery->GetData(pDevInfoStruct, DevInfoStructSize, D3DGETDATA_FLUSH);
```

Comment:
tldr yes it will because there's no GPU involved here.

Reply:
I'd leave that note until (if ever) there's an implementation. Native drivers actually mull over if anything is returned on an initial call with D3DGETDATA_FLUSH, recommending one checks in a loop until results are available. That sounds... like something I'd rather not have done anyway.